messages/teleporter/message_handler.go

```diff
@@ -183,47 +183,63 @@ func (m *messageHandler) ShouldSendMessage(destinationClient vms.DestinationClie
 		return false, nil
 	}
 
-	var decision bool = true
+	deciderRejectedMsg, err := m.deciderRejectedMessage()
+	if err != nil {
+		m.logger.Warn(
+			"Error delegating to decider",
+			zap.String("warpMessageID", m.unsignedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+		)
+	}
```

Comment:
I considered your suggestions but took a step back and decided on something slightly different, in https://github.com/ava-labs/awm-relayer/pull/344/commits/df26aa2b428fc59ec6f8af7e30a2b3d184148c41. Let me know what you think.

Reply:
We should reserve returning pointers for large structs where we want to avoid deep copies. Returning pointers for built in types makes code less readable and more error prone, with no real performance gain.

Reply:
I thought it was also appropriate to use pointers for optional values, like we often do in structs, like I had originally done with the port value in the config struct. It felt weird to me to return a "decision" value while also returning an error, but googling just now I see that the value is generally considered undefined when the error is non-nil, so I guess that weirdness is just normality. :) addressed in https://github.com/ava-labs/awm-relayer/pull/344/commits/7e02e1ac549f556c3d3d9c4327f245232551e7a4